doc/api/http2.md

```diff
@@ -2410,8 +2410,8 @@ changes:
     queued to be sent, and unacknowledged `PING` and `SETTINGS` frames are all
     counted towards the current limit. **Default:** `10`.
   * `maxHeaderListPairs` {number} Sets the maximum number of header entries.
-    This is similar to [`http.Server#maxHeadersCount`][] or
-    [`http.ClientRequest#maxHeadersCount`][]. The minimum value is `4`.
+    This is similar to [`http.Server.maxHeadersCount`][] or
+    [`http.ClientRequest.maxHeadersCount`][]. The minimum value is `4`.
```

Comment:
Not sure this PR is improving the clarity, I think we want to make it clear if those are static properties, on instance getters.
```suggestion
    This is similar to [`http.Server.prototype.maxHeadersCount`][] or
    [`http.ClientRequest.prototype.maxHeadersCount`][]. The minimum value is `4`.
```
but that notation is not ideal either, and might be as confusing. We could try doing this maybe:
```suggestion
    This is similar to [`httpServer.maxHeadersCount`][] or
    [`httpClientRequest.maxHeadersCount`][]. The minimum value is `4`.
```

Reply:
I'm not sure I understand. Is not `maxHeadersCount` a property of a `http.Server` instance and a property of `http.ClientRequest` (ignore the fact that we don't create them via `new`, but we treat them like a class instance)?

Reply:
`http.Server` is not an instance of `http.Server` though.

```console
$ node -p 'http.Server instanceof http.Server'
false
$ node -p '"maxHeadersCount" in http.Server'   
false
$ node -p '"maxHeadersCount" in new http.Server()'                
true
```



> ignore the fact that we don't create them via `new`, but we treat them like a class instance

I don't think that's something we can ignore, it would be very confusing.

Reply:
I agree. The problem is that the # is arbitrary and might also be mistaken in the future for the private class accessor.

What about keeping the `.` and specifically annotate destinations sections to outline they are direct properties of the classes and not of the instances?

Reply:
Don't you think that the following would be way less confusing?

```suggestion
    This is similar to [`httpServer.maxHeadersCount`][] or
    [`httpClientRequest.maxHeadersCount`][]. The minimum value is `4`.
```

Reply:
Honestly not. `http` act as namespace. Collapsing into the object name gives no benefit IMHO.

Reply:
Then I guess `#` is the only acceptable convention, and specifically annotate destinations sections to outline they are public properties, not private.

Reply:
> All are read-write and use the same notation in other pages, like https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#subprocessconnected.

This link doesn't contain any example of an instance property being documented as a static property, does it?

Reply:
That's what I'm saying. Those properties are not static. Why do you think they are static?

Reply:
`http.Server` is a class, so `http.Server.whatever` must be a static property, at least that's how I read it.

Reply:
Oh, now I see the misunderstanding.
If you follow the link, then the section title will be `server.whatever`. So instance method. The upper case will be use only for cross-page linking. I don't think this will confuse readers.

Reply:
That's true, but I think it also applies for the use of `#` (when the user click on the link, it will be clear to them that it is not a private property).
Another upside of `#` over `.` is that `http.Server#whatever` is not valid JS, `http.Server.whatever` is but doesn't mean what we actually mean. I know it would confuse me.
Maybe we can try to use another convention? Like maybe `<http.Server instance>.whatever`?

Reply:
@aduh95 I followed a completely different approach: I use the same title of the destination links but I made explicit in the source that we're referring to another module. What do you think about it?